### ISSUE TITLE
git-flow specific hooks

### ISSUE THREAD

**Issue by Thiana** [nvie/gitflow#60]
Have you considered adding hooks for git-flow?

ie,  `flow-{action} [start|finish]` - Invoked when {action} is started/finished  would be ideal for integrating with an issue tracker to mark a feature/hotfix/etc as assigned/closed.

**Comment by nvie** [comment]
Yeah, there definitely will be hooks for many actions, but I think I'll reserve them for the rewrite. I don't want to add too many things now that will change with the next major release again.

There should be hooks for before/after actions, validating, auto-bumping of version number and many a few more.

**Comment by jankeesvw** [comment]
+1 for this feature!! 

I would use this to upload my project to the staging server on 'git flow feature finish' and upload to the production server with 'git flow release finish'

**Comment by haf** [comment]
+1

**Comment by stangel** [comment]
+1 more!  Is this still on your radar?

**Comment by jamescarr** [comment]
Yeah! I was just having some devs complain the other day that they wish calling "git flow release 1.0.25" would automatically update the version in their package.json for them because it's a release chore. :)

**Comment by petervanderdoes** [comment]
Please check https://github.com/nvie/gitflow/issues/171#issuecomment-3266460 as there is a fork available with hooks implementation. It's a work in progress so I need people to test the living daylights out the fork.

**Comment by isnok** [comment]
good idea! +1

**Comment by nvie** [comment]
Closing all issues as I'm archiving this repo. git-flow lives on at [git-flow-next](https://github.com/gittower/git-flow-next) if you're still using it. Thanks for reporting!
